cmd/api/errors.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"bookshop/internal/utils"
+	"log/slog"
+	"net/http"
+)
+
+func (app *application) errorResponse(w http.ResponseWriter, r *http.Request, status int, message string) {
+	if err := utils.WriteJSON(w, status, message, nil); err != nil {
+		app.logger.Error(
+			"Failed to write response",
+			slog.String("method", r.Method),
+			slog.String("path", r.URL.String()),
+			slog.String("error", err.Error()),
+		)
+		http.Error(w, "Failed to write response.", http.StatusInternalServerError)
```

Comment:
We shouldn't use this function, because we are returning our errors in json format. (refer to go doc to this func, it sets plain/text in headers)

Consider removing http.Error function it does nothing in our case, we are already set correct status in the `writeJSON` method and write into ResponseWriter.